### PR TITLE
Use go 1.17 / Bump dependencies

### PR DESCRIPTION
**Pull request by mrueg**
Use golang 1.17 for build, bump dependencies and tidy up go.sum file.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.16.6 AS builder
+FROM golang:1.17.3 AS builder
```
Comment (xperimental) [inline Dockerfile:1]: There is a newer version by now. Any reason for not using the latest release?

```suggestion
FROM golang:1.17.4 AS builder
```

**Review by xperimental — CHANGES_REQUESTED** [review]
Thanks for the contribution! This was already on my todo for the next days...

Just one small comment.

File: Dockerfile (hunk 1 above)
Comment (mrueg) [reply Dockerfile]: Yes, I missed that .4 was released :D Thanks, adjusted it!

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.16.6 AS builder
+FROM golang:1.17.4 AS builder
```
Comment (xperimental) [inline Dockerfile:1]: I'm sorry, but 1.17.5 was just released. Would you mind updating the PR again?

```suggestion
FROM golang:1.17.5 AS builder
```

**Review by xperimental — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (mrueg) [reply Dockerfile]: done! :)

**Review by xperimental — APPROVED** [review]
Thanks!

**Pull request merged**